employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/SkillService.java

```diff
@@ -0,0 +1,116 @@
+ */
+
+package org.optaweb.employeerostering.service;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.persistence.EntityExistsException;
+import javax.persistence.EntityNotFoundException;
+import javax.transaction.Transactional;
+
+import org.optaweb.employeerostering.domain.Skill;
+import org.optaweb.employeerostering.persistence.SkillRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SkillService extends AbstractRestService {
+
+    private final SkillRepository skillRepository;
+
+    public SkillService(SkillRepository skillRepository) {
+        this.skillRepository = skillRepository;
+    }
+
+    @Transactional
+    public List<Skill> getSkillList(Integer tenantId) {
+        return skillRepository.findAllByTenantId(tenantId);
+    }
+
+    @Transactional
+    public Skill getSkill(Integer tenantId, Long id) {
+        Optional<Skill> skillOptional = skillRepository.findById(id);
+
+        if (!skillOptional.isPresent()) {
+            throw new EntityNotFoundException("No Skill entity found with ID (" + id + ").");
+        }
+
+        validateTenantIdParameter(tenantId, skillOptional.get());
+        return skillRepository.findById(id).get();
+    }
+
+    @Transactional
+    public Boolean deleteSkill(Integer tenantId, Long id) {
+        Optional<Skill> skillOptional = skillRepository.findById(id);
+
+        if (!skillOptional.isPresent()) {
+            throw new EntityNotFoundException("No Skill entity found with ID (" + id + ").");
+        }
+
+        validateTenantIdParameter(tenantId, skillOptional.get());
+        skillRepository.deleteById(id);
+        return true;
+    }
+
+    @Transactional
+    public Skill createSkill(Integer tenantId, Skill skill) {
+        validateTenantIdParameter(tenantId, skill);
+
+        String name = skill.getName();
+
+        //If name is null
+        if (name == null) {
+            throw new IllegalStateException("Skill entity name with tenantId (" + tenantId + ") cannot be null.");
+        }
+
+        List<Skill> skillList = skillRepository.findAllByTenantId(tenantId);
+
+        for (Skill s : skillList) {
+            //If skill already exists
+            if (s.getName().equals(name)) {
+                throw new EntityExistsException("Skill entity with tenantId (" + tenantId + ") and name (" + name
+                        + ") already exists.");
+            }
+        }
+
+        return skillRepository.save(skill);
+    }
+
+    @Transactional
+    public Skill updateSkill(Integer tenantId, Skill skill) {
+        validateTenantIdParameter(tenantId, skill);
+
+        Optional<Skill> skillOptional = skillRepository.findById(skill.getId());
+
+        //If name is null
+        if (skill.getName() == null) {
+            throw new IllegalStateException("Skill entity name with tenantId (" + tenantId + ") cannot be null.");
+        }
+        //Skill not found
+        else if (!skillOptional.isPresent()) {
```

Comment:
Follow the `KIE Java Convention` code style, `else if` should be on the same line as the closing curly brace for preceding if block. (You can use the shortcut `ctl + alt + L` to auto format your file`.